File: src/content/category.ts
import { z, defineCollection, reference } from "astro:content";

const categoryCollection = defineCollection({
  type: "content",
  schema: z.object({
    title: z.string(),
    date: z.date().default(() => new Date()),
    slug: z.string(),
    image: z.string().optional(),
    parentCategory: z.string().optional(),
    displayOrder: z.number().optional(),
    icon: z.string().optional(),
    tags: z.array(z.string()).optional(),
    isVisible: z.boolean().default(() => true),

    SEOTitle: z.string().optional(),
    SEODescription: z.string().optional(),
    SEOKeywords: z.array(z.string()).optional(),
  }),
});

export default categoryCollection;

File: cli/models/category/category.go
package category

import (
	"time"
)

type Category struct {
	Title          string    `yaml:"title"`
	Date           time.Time `yaml:"date,omitempty"`
	Slug           string    `yaml:"slug"`
	Image          *string   `yaml:"image,omitempty"`
	ParentCategory *string   `yaml:"parentCategory,omitempty"`
	DisplayOrder   *int      `yaml:"displayOrder,omitempty"`
	Icon           *string   `yaml:"icon,omitempty"`
	Tags           []string  `yaml:"tags,omitempty"`
	IsVisible      bool      `yaml:"isVisible"`
	SEOTitle       *string   `yaml:"SEOTitle,omitempty"`
	SEODescription *string   `yaml:"SEODescription,omitempty"`
	SEOKeywords    []string  `yaml:"SEOKeywords,omitempty"`
}
